client.go

```diff
@@ -1307,13 +1307,21 @@ func (l *link) muxDetach() {
 		Closed: true,
 		Error:  detachError,
 	}
-	select {
-	case l.session.tx <- fr:
-	case <-l.session.done:
-		if l.err == nil {
-			l.err = l.session.err
+
+Loop:
+	for {
+		select {
+		case l.session.tx <- fr:
+			// after sending the detach frame, break the read loop
+			break Loop
+		case _ = <-l.rx:
+			// discard link frames since we are sending a detach
```

Comment:
Can you change this to something along the lines of "discard incoming frames to avoid blocking session.mux"? I prefer to avoid personal pronouns in code comments.